modules/web/src/com/haulmont/cuba/web/gui/components/WebSuggestionPickerField.java

```diff
@@ -294,7 +296,13 @@ protected void showSuggestions(List<V> suggestions, boolean userOriginated) {
             lastDialog = dialogScreen;
         }
 
-        if (lastDialog == null || Objects.equals(frameOwner, lastDialog)) {
+        boolean frameInsideDialog = false;
+        if(lastDialog instanceof ComponentContainer) {
```

Comment:
`lastDialog` can't be an instance of `ComponentContainer`, you need to obtain its `window` object and check if it is an instance of `ComponentContainer`.

In order to fix the problem, I would recommend getting a _screen_ that contains a fragment and replace the `frameOwner` variable value with it. In this case, we don't need to add additional conditions to the existing check, e.g.:

```
...

// In case the suggestion field is in the fragment,
// we need to get the screen containing it
if (frameOwner instanceof ScreenFragment) {
    frameOwner = ComponentsHelper.getScreen(((ScreenFragment) frameOwner));
}

if (lastDialog == null || Objects.equals(frameOwner, lastDialog)) {
    component.showSuggestions(suggestions, userOriginated);
}
```

Reply:
Sorry, initially problem found at legacy window, which is an instance of `ComponentContainer`. Fixed it, according to recommendations. Works fine both on new and legacy screens.